src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,217 @@
+        }
+    }
+
+    private Node<T> findNodeByIndex(int index) {
+        if (index == 0) {
+            return first;
+        } else if (index == size - 1) {
+            return last;
+        }
+        int count = 0;
+        MyLinkedList.Node<T> currentNode = this.first;
+        while (true) {
+            if (count == index) {
+                break;
+            }
+            currentNode = currentNode.next;
+            count++;
+        }
+        return currentNode;
+    }
+
+    private Node<T> findNodeByValue(T t) {
+        if (size == 1) {
+            return first;
+        }
+        MyLinkedList.Node<T> currentNode = this.first;
+        int count = 0;
+        boolean found = false;
+        while (count < size - 1) {
+            if (currentNode.element == t || currentNode.element.equals(t)) {
+                found = true;
+                break;
+            }
+            currentNode = currentNode.next;
+            count++;
+        }
+        if (found) {
+            return currentNode;
+        }
+        return null;
+    }
+
+    private void linkFirst(T t) {
+        MyLinkedList.Node<T> first = this.first;
+        MyLinkedList.Node<T> newNode = new MyLinkedList.Node<T>(null, t, first);
+        if (first == null) {
+            this.first = newNode;
+            this.last = newNode;
+        } else {
+            first.previous = newNode;
+            this.first = newNode;
+        }
+        size++;
+    }
+
+    private void unLinkFirst() {
+        if (this.first.next == null) {
+            this.first = null;
+            this.last = null;
+            size = 0;
+            return;
+        }
+        this.first = this.first.next;
+        size--;
+    }
+
+    private void linkLast(T t) {
+        MyLinkedList.Node<T> last = this.last;
+        MyLinkedList.Node<T> newNode = new MyLinkedList.Node<T>(last, t, null);
+        if (last == null) {
+            this.first = newNode;
+            this.last = newNode;
+        } else {
+            last.next = newNode;
+            this.last = newNode;
+        }
+        size++;
+    }
+
+    private void unLinkLast() {
+        if (this.last.previous == null) {
+            this.first = null;
+            this.last = null;
+            size = 0;
+            return;
+        }
+        this.last = this.last.previous;
+        size--;
+    }
+
+    private void unlink(Node<T> node) {
+        MyLinkedList.Node<T> leftNode = node.previous;
+        MyLinkedList.Node<T> rightNode = node.next;
+        leftNode.next = rightNode;
+        rightNode.previous = leftNode;
+        size--;
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        linkLast(value);
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == 0) {
+            linkFirst(value);
+            return;
+        } else if (index == size) {
+            linkLast(value);
+            return;
+        }
```

Comment:
if (index == size) {
            linkLast(value);
            return;
        }

Reply:
done. I had a question, but i answered it myself:)